mlx/random.cpp

```diff
@@ -192,6 +193,62 @@ array normal(
   return samples;
 }
 
+array multivariate_normal(
+    const array& mean,
+    const array& cov,
+    const std::vector<int>& shape,
+    Dtype dtype,
+    const std::optional<array>& key,
+    StreamOrDevice s) {
+  auto stream = to_stream(s);
+
+  if (dtype != float32) {
+    throw std::invalid_argument("[multivariate_normal] dtype must be float32.");
+  }
+
+  auto n = *(mean.shape().end() - 1);
+
+  // Check shapes comatibility of mean and cov
+  if (n != *(cov.shape().end() - 2)) {
+    throw std::invalid_argument(
+        "[multivariate_normal] last two dimensions of cov must be equal.");
```

Comment:
This error message doesn't match the condition. Do you mean to check `cov.shape(-1) == cov.shape(-2)`?

Reply:
Yes!